File: WebMacroSoftKeyboard.PluginContract/ConfigValue.cs
namespace WebMacroSoftKeyboard.PluginContract
{
    public class ConfigValue
    {
        public ConfigValue(string symbolicName, object value)
        {
            SymbolicName = symbolicName;
            Value = value;
        }

        public string SymbolicName { get; }

        public object Value { get; }
    }
}

File: WebMacroSoftKeyboard/ClientApp/src/app/Models/ConfigValue.ts
import { Injectable } from "@angular/core";
import { Adapter } from "./Adapter";

export class ConfigValue
{
  constructor(
    public symbolicName: string,
    public value: any) { }
}


@Injectable({
  providedIn: "root",
})
export class ConfigValueAdapter implements Adapter<ConfigValue>
{
  adapt(item: any): ConfigValue
  {
    return new ConfigValue(item.symbolicName, item.value);
  }
}
